### DISCUSSION TITLE
Asset modification mark in the UI

### DISCUSSION THREAD

**Discussion by dridk** [dagster-io/dagster#18618] [Q&A] [answered]
what about displaying on the dagit asset if the source code has been modified since the last materialization?

- you can store the hash of the asset for each materialization

- if the hash change, display on the box "asset has change" 

This would be very useful. Because, for a success materialized asset, we can't know if we'll be able to rematerialize it in the same way.

**Comment by sryza** [comment, marked as answer]
Hey @dridk - if you set the `code_version` on the asset - either automatically or manually - Dagster will display a label saying that it has changed.  However, there's no universally accepted way of determining whether Python source code has changed - e.g. what if the body of the compute function stays the same, but it calls out to a different function that changed? Or uses a library that changed?

**Reply by dridk** [reply]
Humm... That's true. 
So, I guess code_version is a good solution. I can use my own hash if I want. 

Thank you
